digits/model/views.py

```diff
@@ -205,30 +195,54 @@ def to_pretrained(job_id):
 
     # Set defaults:
     labels_path = None
+    mean_path = None
     resize_mode = None
 
     if "labels file" in info:
         labels_path = os.path.join(task.dataset.dir(),info["labels file"])
+    if "mean file" in info:
+        mean_path = os.path.join(task.dataset.dir(),info["mean file"])
     if "image resize mode" in info:
         resize_mode = info["image resize mode"]
 
+    model_file = os.path.join(job.dir(),str(task.model_file))
+
+    # If jobs don't container model_file (too old), raise exception:
+    if not os.path.isfile(model_file):
+       raise werkzeug.exceptions.BadRequest('Model file not found in job dir. Job may be too old for conversion.')
+
     job = PretrainedModelJob(
         snapshot_filename,
-        job.dir() + "/" + task.model_file ,
+        model_file,
         labels_path,
+        mean_path,
         info["framework"],
         info["image dimensions"][2],
         resize_mode,
         info["image dimensions"][0],
         info["image dimensions"][1],
-        username = auth.get_username(),
+        username = username,
         name = info["name"]
     )
 
     scheduler.add_job(job)
+    return job
+
+@blueprint.route('/<job_id>/to_pretrained',methods=['GET', 'POST'])
+def to_pretrained(job_id):
+    epoch = -1
+    # GET ?epoch=n
+    if 'epoch' in flask.request.args:
+        epoch = float(flask.request.args['epoch'])
+
+    # POST ?snapshot_epoch=n (from form)
+    elif 'snapshot_epoch' in flask.request.form:
+        epoch = float(flask.request.form['snapshot_epoch'])
 
-    return flask.redirect(flask.url_for('digits.views.home',tab=3)), 302
+    username = auth.get_username()
 
+    create_pretrained_model(job_id,username,epoch)
```

Comment:
Quite often the pretrained model job hasn't completed by the time we get redirected to the home page and therefore we don't see the new job unless we refresh the home page. Can we add a call to `job.wait_completion()` before the redirection?